### PR TITLE
My Site Dashboard: Refactor dashboard architecture and data source

### LINKED ISSUE #18208: My Site Dashboard: while displaying two posts cards, removing all posts causes odd visual behavior
### Expected behavior

1. When removing all drafts or scheduled posts and 2 cards are being displayed
2. One of them would be hidden

### Actual behavior

We show the next post prompt. Only after a subsequent update, the card disappears. Here's an example:

https://user-images.githubusercontent.com/25306722/159933106-e2646ae7-a924-498a-a0b3-0593a1da21c9.mp4

### Steps to reproduce the behavior

1. Have both drafts and scheduled displayed
2. Go to posts list
3. Trash the scheduled post or the draft post (or all)
4. Go back to the dashboard

### LINKED ISSUE #18308: My Site Dashboard: Refactor dashboard datasource
## Description

Refactor the way the dashboard is populated. Right now posts cards are all embedded in one cell, and the cell is responsible for figuring out what to display and adding the cards. The aim is to change this so that the data source (BlogDashboardService) returns a list of cards ready to display.

Ref: p1648002837904629-slack-G017Y852K24
Ref: https://github.com/wordpress-mobile/WordPress-iOS/issues/18228#issuecomment-1086998412

## Execution Plan
All the next steps will target a feature branch and not trunk since we haven't decided if we'll go through with this or not.
- [x] Reduce sections usage. Only two sections instead of one for each card. One for quick actions, and one for the rest of the cards. 
    - #18309
- [x] Refactor snapshot's logic so that the service returns a list of ALL cards to be shown.
    - #18310 
    - #18311
- [x] Update `DashboardPostsCardCell` to only show one card. This cell will probably be split into two cells.
    - By this step, the refactor should be wrapped up and the dashboard should behave the exact same way it used to do before.
    - #18340
    - #18351
- [x] ~~Add a new class responsible for tracking the cards shown and firing analytics events.~~
    - Related issue: #18228

### LINKED ISSUE #18355: My Site Dashboard: Failing to load posts error not shown
### Expected behavior
Error message to be displayed inside posts cards when loading posts fails

### Actual behavior
Next post view is displayed instead

### Steps to reproduce the behavior
1. Replace the body of `PostService.syncPostsOfType` at PostService.m line 197 with:
```
double delayInSeconds = 1;
dispatch_time_t time = dispatch_time(DISPATCH_TIME_NOW, (int64_t)(delayInSeconds * NSEC_PER_SEC));    dispatch_after(time, dispatch_get_main_queue(), ^{
    NSError *error = [NSError errorWithDomain:@"" code:1 userInfo:nil];
    failure(error)
});
```
This simulates posts loading failure.

2. Do a fresh install to clear cached posts data
3. Open a site that has drafts and scheduled posts
4. Open the dashboard
5. Observe that the next post view is displayed instead of the error message

### PR DESCRIPTION
**Pull request by hassaanelgarem**
Fixes #18355
Fixes #18208
Closes #18308

## Description
This PR refactors how the dashboard displays the returned cards. Posts cards are significantly affected, as we now use a cell for each card instead of one cell for all cards.
The PR combines work done in the following PRs:
- #18309
- #18310
- #18311 
- #18340 
- #18351 
- #18391

## Testing Instructions

Testing isn't really needed as all changes have been tested in their corresponding PRs.

## Regression Notes
1. Potential unintended areas of impact
Dashboard, especially posts cards

3. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and unit tests, when applicable

5. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 19.8
 -----
-
+* [*] [internal] My Site Dashboard: Made some changes to the code architecture of the dashboard. The majority of the changes are related to the posts cards. It should have no visible changes but could cause regressions. [#18405]
```
Comment (momo-ozawa) [inline RELEASE-NOTES.txt]: 🎉

**Review by momo-ozawa — APPROVED** [review]
@hassaanelgarem Thank you for tackling this refactor!

File: RELEASE-NOTES.txt (hunk 1 above)
Comment (leandroalonso) [reply RELEASE-NOTES.txt]: I think it worth mentioning: "please test it by creating drafts and scheduled posts and testing that they appear on the dashboard".

**Review by leandroalonso — APPROVED** [review]

**Comment by hassaanelgarem** [comment]
@dvdchr FYI: This makes some changes to the dashboard's architecture. It shouldn't affect the blogging prompt card though.
It might be useful to take a look at the changes in `DashboardCard` for future reference.

File: RELEASE-NOTES.txt (hunk 1 above)
Comment (hassaanelgarem) [reply RELEASE-NOTES.txt]: Added in 2b86889a42044be4a883210ead296a3e34f10044

**Pull request merged**